### DISCUSSION TITLE
I am struggling to get this to work. Please help!

### DISCUSSION THREAD

**Discussion by EspeeFunsail** [klaasnicolaas/home-assistant-glow#94] [Q&A]
I am new to coding and electronic tinkering and I have been struggling very hard to get this to work. I have all the hardware connected as described and linked to Home Assistant. I get readings back, but it either doesn't make sense or I don't get anything at all. Also, when I have it connected, the LED flashes red sporadically. The sensor itself does not seem to have an effect on when it pulses. I have tried turning the dail on the sensor to maybe adjust the sensitivity, but also just give random output.

I will really appreciate any assistance. I feel like I will never get this to work and just need someone to maybe show me what I might be missing.

https://imgur.com/a/Qk3wLjN Video showing my setup

Thank you

**Comment by klaasnicolaas** [comment]
Have you checked if the polarity of your photodiode is correct? There are some bad batches in circulation that have the LED reversed. 
https://github.com/klaasnicolaas/home-assistant-glow#wrong-soldered-diode

**Comment by EspeeFunsail** [comment]
Yes, it actually came in the wrong orientation and I had to desolder and
invert it to be correct. So that is not the problem.

Anything else you can think of?

**Reply by klaasnicolaas** [reply]
But your video is the old situation, it clearly shows that something goes wrong with the photodiode board. So what do you still run into after reversing the diode?

**Reply by EspeeFunsail** [reply]
The video was taken with the photodiode in the correct orientation. That is why I don't know why it's not working.

**Reply by EspeeFunsail** [reply]
Hi, I tried another sensor board and it seems to be working now. Might have been something wrong with the first one.

**Reply by klaasnicolaas** [reply]
So far I can see on the video the polarity is wrong 😉
